app/models/processor.rb

```diff
@@ -0,0 +1,81 @@
+require 'json'
+require 'useragent'
+require 'uri'
+
+module Processor
+extend self
+
+  def clean_data(params)
+    {identifier: params[:identifier], root_url: params[:rootUrl]}
+  end
+
+  def json_parser(params)
+    JSON.parse(params)
+  end
+
+  def params_cleaner(params)
+    user_agent, referredby, url = params_parser(params)
+
+    {
+     requested_at: json_parser(params)['requestedAt'],
+     responded_in: json_parser(params)['respondedIn'],
+     referred_by_id: ReferredBy.find_or_create_by(root_url: referredby.host, path: referredby.path).id,
+     request_type_id: RequestType.find_or_create_by(http_verb: json_parser(params)['requestType']).id,
+     event_id: Event.find_or_create_by(event_name: json_parser(params)['eventName']).id,
```

Comment:
notice you're calling json_parser(params) in this method quite a few times. with every call, you're re-parsing the JSON. is this necessary? we could be more efficient if we just parsed this JSON one time, then used it repeatedly

also - this method is super long. can we move each of these ActiveRecord calls (find_or_create_by) into separate methods?